### PR TITLE
Per Joint Loss Function, Bi-Direction Encoder Additional Args, Loss Calculation

### PR DESCRIPTION
**Pull request by jrevansjr**

### CODE REVIEW

File: utils/loss.py [hunk 1]
```diff
@@ -17,11 +44,11 @@ def compute_validation_loss(args, model, datasets, criterion, device, mask):
         mask: The mask for masked objective (optional)
 
     Returns:
-        A tuple of the loss and the val_iter
+        A tuple of the loss and the batch_size
     """
     epoch_val_loss = 0
```
Comment (moduIo) [inline utils/loss.py:49]: Do we need model.eval()? And if we do, we need a way to switch it back to model.train()
